### DISCUSSION TITLE
Using <Cotroller /> for object fields

### DISCUSSION THREAD

**Discussion by Melancholism** [react-hook-form/react-hook-form#11463] [Q&A] [answered]
I have a simple form with default values set as follows:
```js
  const methods = useForm({
    defaultValues: {
      car: {
        name: "",
        model: "",
      },
    },
  });
```
My component renders two inputs using `<Controller/>`. The first input is supposed to be an MUI Autocomplete (for the sake of simplicity, it renders nothing but a single `div` in this [CSB](https://codesandbox.io/p/sandbox/register-forked-x3w5v8?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cls23y0310006396gau686bpy%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cls23y0300002396g2b5ylln2%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cls23y0300003396gm520a897%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cls23y0300005396gy8orh6rc%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cls23y0300002396g2b5ylln2%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls23y02z0001396gnbzyka1u%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cls23y0300002396g2b5ylln2%2522%252C%2522activeTabId%2522%253A%2522cls23y02z0001396gnbzyka1u%2522%257D%252C%2522cls23y0300005396gy8orh6rc%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls23y0300004396g5etnlkut%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522cls23y0300005396gy8orh6rc%2522%252C%2522activeTabId%2522%253A%2522cls23y0300004396g5etnlkut%2522%257D%252C%2522cls23y0300003396gm520a897%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522cls23y0300003396gm520a897%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)) that sets the whole car object (`name="car"`). and the second one simply changes the car's name (`name="car.name"`). The issue is when I set the car value in the first input, the second one doesn't show the set car name.

I simulated this process by rendering a simple button that sets the `car object` value using the `setValue` function here in this [CSB](https://codesandbox.io/p/sandbox/register-forked-x3w5v8?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cls23y0310006396gau686bpy%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cls23y0300002396g2b5ylln2%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cls23y0300003396gm520a897%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cls23y0300005396gy8orh6rc%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cls23y0300002396g2b5ylln2%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls23y02z0001396gnbzyka1u%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cls23y0300002396g2b5ylln2%2522%252C%2522activeTabId%2522%253A%2522cls23y02z0001396gnbzyka1u%2522%257D%252C%2522cls23y0300005396gy8orh6rc%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls23y0300004396g5etnlkut%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522cls23y0300005396gy8orh6rc%2522%252C%2522activeTabId%2522%253A%2522cls23y0300004396g5etnlkut%2522%257D%252C%2522cls23y0300003396gm520a897%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522cls23y0300003396gm520a897%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D).
The issue gets fixed when I change the first input's name to `car.model`. But in the case that the first input's name is `car`, pressing the setter button doesn't set the car name text input. Am I doing anything wrong here? Am I not supposed to achieve my goal this way? Any help would be appreciated a lot. Thanks in advance.

**Comment by sujinleeme** [comment, marked as answer]
It would help if you fix `handleSetFields` to update fields individually to make each controller see the changes. 
`<Controller name="car.model">` doesn't know `car` object has changed. It only detects field changes.

```ts
  const handleSetFields = () => {
    setValue("car.name", "Ford")
    setValue("car.model", "Mustang");
  };
```

**Reply by Melancholism** [reply]
Oh! I see!

> It only detects its field changes. 

That's the point I was missing.

Thank you for your time. Appreciate it.
